src/webviews/mongoClusters/collectionView/components/toolbar/ToolbarViewNavigation.tsx

```diff
@@ -6,37 +6,99 @@
+
     const [currentContext, setCurrentContext] = useContext(CollectionViewContext);
 
     function goToNextPage() {
+        const newPage = currentContext.currrentQueryDefinition.pageNumber + 1;
+
         setCurrentContext({
             ...currentContext,
             currrentQueryDefinition: {
                 ...currentContext.currrentQueryDefinition,
-                pageNumber: currentContext.currrentQueryDefinition.pageNumber + 1,
+                pageNumber: newPage,
             },
         });
+
+        trpcClient.common.reportEvent
+            .mutate({
+                eventName: 'pagination',
+                properties: {
+                    source: 'next-page',
+                    ui: 'button',
+                    view: currentContext.currentView,
+                },
+                measurements: {
+                    page: newPage,
+                    pageSize: currentContext.currrentQueryDefinition.pageSize,
+                },
+            })
+            .catch((error) => {
+                console.error('Failed to report pagination event:', error);
+            });
     }
 
     function goToPreviousPage() {
+        const newPage = Math.max(1, currentContext.currrentQueryDefinition.pageNumber - 1);
+
         setCurrentContext({
             ...currentContext,
             currrentQueryDefinition: {
                 ...currentContext.currrentQueryDefinition,
-                pageNumber: Math.max(1, currentContext.currrentQueryDefinition.pageNumber - 1),
+                pageNumber: newPage,
             },
         });
+
+        trpcClient.common.reportEvent
+            .mutate({
+                eventName: 'pagination',
+                properties: {
+                    source: 'prev-page',
+                    ui: 'button',
+                    view: currentContext.currentView,
+                },
+                measurements: {
+                    page: newPage,
+                    pageSize: currentContext.currrentQueryDefinition.pageSize,
+                },
+            })
+            .catch((error) => {
+                console.error('Failed to report pagination event:', error);
+            });
     }
 
     function goToFirstPage() {
         setCurrentContext({
             ...currentContext,
             currrentQueryDefinition: { ...currentContext.currrentQueryDefinition, pageNumber: 1 },
         });
+
+        trpcClient.common.reportEvent
+            .mutate({
+                eventName: 'pagination',
+                properties: {
+                    source: 'first-page',
+                    ui: 'button',
+                    view: currentContext.currentView,
+                },
+                measurements: {
+                    page: 1,
+                    pageSize: currentContext.currrentQueryDefinition.pageSize,
+                },
+            })
+            .catch((error) => {
+                console.error('Failed to report pagination event:', error);
```

Comment:
If this is error then either it should show message to user or send to telemetry. If this is debug info, please use console.debug or log.